packages/flutter/lib/src/widgets/two_pane.dart

```diff
@@ -0,0 +1,482 @@
+///
+/// If not provided, [textDirection] defaults to the ambient [Directionality].
+/// If none are provided, then the widget asserts during build in debug mode.
+/// The resolved [direction] of TwoPane is [Axis.horizontal] when there are
+/// display features separating the screen into two horizontal sub-screens, in
+/// which case the provided [direction] is ignored.
+///
+/// See also
+///
+///  * [DisplayFeature] and [MediaQueryData.displayFeatures], to further
+///    understand display features
+///  * [MediaQueryData.removeDisplayFeatures] which is used to remove padding,
+///    insets and display features for each pane.
+class TwoPane extends StatelessWidget {
+  /// Create a layout that shows two pane widgets side by side.
+  const TwoPane({
+    Key? key,
+    required this.startPane,
+    required this.endPane,
+    this.paneProportion = 0.5,
+    this.textDirection,
+    this.verticalDirection = VerticalDirection.down,
+    this.direction = Axis.horizontal,
+    this.panePriority = TwoPanePriority.both,
+    this.inset = EdgeInsets.zero,
+  }) : super(key: key);
+
+  /// The first pane.
+  ///
+  /// On a horizontal layout, where [direction] is [Axis.horizontal]:
+  ///
+  ///   * The first pane is on the left when [textDirection] is
+  ///     [TextDirection.ltr]
+  ///   * The first pane is on the right when [textDirection] is
+  ///     [TextDirection.rtl]
+  ///
+  /// On a vertical layout, where [direction] is [Axis.vertical]:
+  ///
+  ///   * The first pane is at the top when [verticalDirection] is
+  ///     [VerticalDirection.down]
+  ///   * The first pane is at the bottom when [verticalDirection] is
+  ///     [VerticalDirection.up]
+  ///
+  /// If [panePriority] is [TwoPanePriority.start], this is the only pane
+  /// visible.
+  final Widget startPane;
+
+  /// The second pane.
+  ///
+  /// On a horizontal layout, where [direction] is [Axis.horizontal]:
+  ///
+  ///   * The second pane is on the right when [textDirection] is
+  ///     [TextDirection.ltr]
+  ///   * The second pane is on the left when [textDirection] is
+  ///     [TextDirection.rtl]
+  ///
+  /// On a vertical layout, where [direction] is [Axis.vertical]:
+  ///
+  ///   * The second pane is at the bottom when [verticalDirection] is
+  ///     [VerticalDirection.down]
+  ///   * The second pane is at the top when [verticalDirection] is
+  ///     [VerticalDirection.up]
+  ///
+  /// If [panePriority] is [TwoPanePriority.end], this is the only pane
+  /// visible.
+  final Widget endPane;
+
+  /// Proportion of the available space occupied by the first pane. The second
+  /// pane takes over the rest of the screen.
+  ///
+  /// A value of 0.5 will make the 2 panes equal.
+  ///
+  /// This property is ignored is the screen is split into sub-screens by a
+  /// [DisplayFeature], in which case each pane takes over one sub-screen.
+  final double paneProportion;
+
+  /// Same as [Flex.textDirection].
```

Comment:
You could template some docs from `Flex` for these following parameters and use a macro. Just saying they are the same isn't super helpful if folks don't know what they are.